### ISSUE TITLE
Match text to language typed in / possible preferences of languages to search from

### ISSUE THREAD

**Issue by Ethan1978** [Rascalov/Anki-Simple-Forvo-Audio#18]
Possible enhancement? Forvo matches text in search and gives results based on language used, would it be possible to do something similar or tell it what deck is what language, so that I never have to select the language manually.

**Comment by Rascalov** [comment]
Welcome to the repo!
If I understand you correctly, depending on the deck you want to add flashcards to, the target language used to search would be different? 
![image](https://github.com/Rascalov/Anki-Simple-Forvo-Audio/assets/31393063/049c12f1-b502-4cef-8b17-d7f25961ed8f)
I currently only set the default target language to the one that was last used. I can see why this would be useful for people that learn multiple languages.

However, doing this would require a unique property on the deck that the Forvo addon would be able to identify and link to a language. Maybe a language code inside the deck's name (e.g.: "[ru]MyDeck"). 
Currently I don't use or interact with the deck in this function, so I would need some time to figure the best way to do so. 
The downside of this approach, is that it would require users to edit the names of their decks to contain or (preferably) start with a language code.

Let me know your thoughts on this.

**Comment by Ethan1978** [comment]
Yes you are correct (reference to image) but could you rather than change the deck name place it in the deck description; the deck description is mainly used for decks uploaded to the anki web anyways so, If it were possible I feel as though this would be better.
I don’t know much about code so I look forward to hearing back from you.
Just a shame that you can’t tag anki decks themselves with a specific code in anki it would make something like this much easier.
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Rascalov** [comment]
Interesting, I did not know a deck description existed, thanks for the heads up.
From that, I can kinda imagine the flow:
1. Get the current deck you're adding cards to
2. if the description has a language code in brackets, it will select the associated language
  2a. if not, after the first audio insertion, add the target language's **language code** to the description **if** the same language code is not already present in the description.

Should not be too hard if I can access the deck from the audio search menu, I might push an update for it today. Let me know if I missed anything in the flow

**Comment by Ethan1978** [comment]
Sounds great didn’t think about adding language code automatically; I’m not really familiar with flow but it seems to look good.
Look forward to an update.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Rascalov** [comment]
It is done, I have added a setting as shown below that will set the language code in the  description of the deck you add cards to.
If you have added an audio, it tags your deck with the language you chose while adding that audio.
Let me know if it works as to your liking. You can update the extension by getting the addon code on [the site](https://ankiweb.net/shared/info/560814150) or cloning the repository to your addon directory. 

![image](https://github.com/Rascalov/Anki-Simple-Forvo-Audio/assets/31393063/458eafa9-b81d-4d72-addf-ea89d7197ee6)

**Comment by Ethan1978** [comment]
Definitely be using this tomorrow, looking forward to it !


Galaxyから送信


-------- 元のメッセージ --------
送信元： Tim ***@***.***>
日時: 2023/06/05 21:24 (GMT+00:00)
宛先： Rascalov/Anki-Simple-Forvo-Audio ***@***.***>
Cc： 鈴木 ***@***.***>, Author ***@***.***>
件名: Re: [Rascalov/Anki-Simple-Forvo-Audio] Match text to language typed in / possible preferences of languages to search from (Issue #18)


It is done, I have added a setting as shown below that will set the language code in the description of the deck you add cards to.
If you have added an audio, it tags your deck with the language you chose while adding that audio.
Let me know if it works as to your liking. You can update the extension by getting the addon code on the site<https://ankiweb.net/shared/info/560814150> or cloning the repository to your addon directory.

[image]

**Comment by Rascalov** [comment]
Very good, let me know if it works as intended for you, after confirmation I can close this issue as resolved

**Comment by Ethan1978** [comment]
Added the settings as you said:

{

"LastSelectedField": "Back",

"LastSelectedLanguage": "Korean_ko",

"MaxForvoDownloads": 100,

"Remember language on a per deck basis": "True",

"audioFileExtension": "mp3",

"downloadPath": "C:\\Users\\ekrug\\AppData\\Roaming\\Anki2\\鈴木\\collection.media",

"ignorePunctuation": "True"
}
***@***.***
It works and edits the description on a single deck, but when I used it on the Korean deck it just added the Korean language code to the Japanese deck.
***@***.***
Japanese deck description
***@***.***
Korean deck description
I hope this is an easy fix because it seems to look very promising; Just wondering why you need to remember last deck used though if It remembers it better this way ?
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Rascalov** [comment]
Does this happen when you edit cards in the browse menu?
Could be that the current deck is not correctly selected in that menu. I will have to fix that in any case

**Comment by Rascalov** [comment]
I pushed out an update for editing the card in the browse window, I think this caused issues as it would update the last deck you added cards to, and the Browse editor does it on a card by card basis. 
 
Please update your Anki and check if the issue is resolved

**Comment by Ethan1978** [comment]
I have updated the addon, it works really well.
But I removed the Japanese language code from the deck’s description just to see if it would put it back in; however the language code didn’t re-appear when I added a word’s audio using the addon, wondering if this is intentional?
And just thinking about this working with sub-decks? Don’t know how you would go about this but wondering what you’re thoughts might be since they don’t appear to have a description of any-kind.
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
